wpilibc/src/main/native/include/frc/PneumaticsControlModule.h

```diff
@@ -62,12 +60,28 @@ class PneumaticsControlModule
 
   void UnreserveSolenoids(int mask) override;
 
-  void InitSendable(wpi::SendableBuilder& builder) override;
+  bool ReserveCompressor() override;
+
+  void UnreserveCompressor() override;
+
+  Solenoid MakeSolenoid(int channel) override;
+  DoubleSolenoid MakeDoubleSolenoid(int forwardChannel,
+                                    int reverseChannel) override;
+  Compressor MakeCompressor() override;
 
  private:
+  class DataStore;
+  friend class DataStore;
+  friend class PneumaticsBase;
+  PneumaticsControlModule(HAL_CTREPCMHandle handle, int module);
+
+  static std::shared_ptr<PneumaticsBase> GetForModule(int module);
+
+  std::shared_ptr<DataStore> m_dataStore;
+  HAL_CTREPCMHandle m_handle;
   int m_module;
-  hal::Handle<HAL_CTREPCMHandle> m_handle;
-  uint32_t m_reservedMask;
-  wpi::mutex m_reservedLock;
+
+  static wpi::mutex m_handleLock;
+  static wpi::DenseMap<int, std::weak_ptr<DataStore>> m_handleMap;
```

Comment:
Should we be concerned about a static init race here?

Reply:
Oh, yeah potentially. Thats more often a HAL issue but can trigger here as well. I'll fix it.